cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultAppConfigurer.java

```diff
@@ -63,6 +68,7 @@
   private final Map<String, SparkSpecification> sparks = Maps.newHashMap();
   private final Map<String, WorkflowSpecification> workflows = Maps.newHashMap();
   private final Map<String, ServiceSpecification> services = Maps.newHashMap();
+  private final Map<String, ScheduleConfigurer> schedules = Maps.newHashMap();
```

Comment:
This seems inconsistent with the other members above. Can schedules also be made into ``Map<String, ScheduleSpecification>``. That way, the ``addSchedules`` method would populate a ``Map<String, ScheduleSpecification>`` like the other ``add*`` methods and you won't need the ``createScheduleSpecification`` method to iterate over a ``Map<String, ScheduleConfigurer>`` and convert it into a ``Map<String, ScheduleSpecification>``.